Sources/Workspace/InitPackage.swift

```diff
@@ -49,80 +30,223 @@ public final class InitPackage {
+        self.configPath = configPath
         self.destinationPath = destinationPath
-        self.pkgname = name
-        self.moduleName = name.spm_mangledToC99ExtendedIdentifier()
+        self.mode = mode
+        self.packageName = packageName
+        
+        // These are only needed in the event that --type was not used when creating a package
+        // otherwise if --type is used packageType will not be nill
+        switch (packageType, mode){
+        case (.some(let type), _):
+            self.packageType = type
+        case (.none, .initialize):
+            self.packageType = .library
+        case (.none, .create):
+            self.packageType = .executable
+        }
+    
+        self.packageTemplateName = packageTemplateName
+    }
+
+    public func makePackage() throws {
+        switch Self.createPackageMode {
+        case .new:
+            try makePackageNew()
+        case .legacy:
+            try makePackageLegacy()
+        }
+    }
+    
+    private func makePackageNew() throws {
+        let templateHomeDirectory = configPath.appending(components: "templates", "new-package")
+        var template: String?
+        
+        if let templateName = packageTemplateName {
+            guard fileSystem.exists(templateHomeDirectory.appending(component: templateName)) else {
+                throw InternalError("Could not find template folder: \(templateHomeDirectory.appending(component: templateName))")
+            }
+            
+            template = templateName
+        } else {
+            // Checking if a default template is present
+            if fileSystem.exists(templateHomeDirectory.appending(components: "default")) {
+                template = "default"
+            }
+        }
+        
+        if let templateName = template {
+            try fileSystem.getDirectoryContents(templateHomeDirectory.appending(component: templateName)).forEach {
+                progressReporter?("Copying \($0)")
+                try copyTemplate(fileSystem: fileSystem,
+                                 sourcePath: templateHomeDirectory.appending(components: templateName, $0),
+                                 destinationPath: destinationPath,
+                                 name: packageName)
+            }
+        } else {
+            let packageTemplate = getSwiftPMDefaultTemplate(packageType: packageType)
+            try writePackageStructure(template: packageTemplate)
+        }
+    }
+    
+    private func makePackageLegacy() throws {
+        let template = PackageTemplate(sourcesDirectory: RelativePath("./Sources"),
+                                       testsDirectory: RelativePath("./Tests"),
+                                       createSubDirectoryForModule: true,
+                                       packageType: packageType)
+        
+        try writePackageStructure(template: template)
+    }
+    
+    private func getSwiftPMDefaultTemplate(packageType: InitPackage.PackageType,
```

Comment:
this function is only every called from L152 and always in ".new" mode, so I think you can remove it and just construct the default template in L152